client/js/app/components/explorer/query_builder/extraction_properties_filter.js

```diff
@@ -0,0 +1,45 @@
+var React = require('react');
+var ReactMultiSelect = require('../../common/react_multi_select.js');
+var ExplorerActions = require('../../../actions/ExplorerActions');
+
+var ExtractionPropertiesFilter = React.createClass({
+
+  _getKeys: function() {
+    var keys = _.keys(this.props.result);
+    var keyList = _.map(keys, function(key) {
+      if (typeof this.props.result[key] === "object") {
+        return this._getKey(key, this.props.result[key]);
+      }
+
+      return key;
+    }.bind(this));
+
+    return _.flatten(keyList);
+  },
+
+  _getKey: function(prevKey, obj) {
+    var keys = _.keys(obj);
+
+    return _.map(keys, function(key) {
+      var keyStr = prevKey + '.' + key;
+      if (typeof obj[key] === 'object') {
+        return this._getKey(keyStr, obj[key]);
+      }
+
+      return keyStr;
+    }.bind(this));
+  },
+
+  render: function() {
```

Comment:
Given that this component just wraps `ReactMultiSelect` and has two helper functions for formatters the props to that component, I think this component isn't required. We can format the keys as necessary inside `ExtractionOptions` and move the two functions here, `_getKeys` and `_getKey` into a utility module of some kind (or maybe simply inside `ExtractionOptions`).